common/djangoapps/enrollment/views.py

```diff
@@ -204,6 +204,52 @@ def get(self, request, course_id=None, username=None):
                 }
             )
 
+class EnrollmentUserRolesView(APIView):
+    """
+    **Use Case**
+
+        Get the roles for the current logged-in user.
+
+    **Example Requests**
+
+        GET /api/enrollment/v1/roles/
+
+    **Response Values**
+
+        If the request is successful, an HTTP 200 "OK" response is
+        returned along with a collection of user roles for the
+        logged-in user
+
+    """
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser,
+                              EnrollmentCrossDomainSessionAuth,)
+    permission_classes = ApiKeyHeaderPermissionIsAuthenticated,
+    throttle_classes = EnrollmentUserThrottle,
+
+    @method_decorator(ensure_csrf_cookie_cross_domain)
+    def get(self, request):
+        """
+        Gets a list of all roles for the currently logged-in user
+        """
+        try:
+            roles_data = api.get_user_roles(request.user.username)
+        except Exception:
+            return Response(
```

Comment:
You probably want to write one more test to cover this block of code.  I can show you how to force an exception in the right place using mock.